dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/CheckinContentActionletTest.java

```diff
@@ -0,0 +1,150 @@
+package com.dotmarketing.portlets.workflows.actionlet;
+
+import com.dotcms.contenttype.model.field.*;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.datagen.*;
+import com.dotcms.exception.ExceptionUtil;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.Permissionable;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.DotLockException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.model.ContentletDependencies;
+import com.dotmarketing.portlets.contentlet.model.ContentletVersionInfo;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionClassParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+import com.google.common.collect.ImmutableList;
+import com.liferay.portal.model.User;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Map;
+
+import static com.dotcms.util.CollectionsUtils.map;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(DataProviderRunner.class)
+public class CheckinContentActionletTest {
+
+    private static class TestCase {
+        Contentlet contentlet;
+        User user;
+
+        boolean hasWritePermission;
+
+        public TestCase(Contentlet contentlet, User user, boolean hasWritePermission) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5365897)